code/modules/mob/living/carbon/human/species_types/plasmamen.dm

```diff
@@ -51,12 +51,13 @@
 					if(H.on_fire && H.fire_stacks > 0)
 						H.adjust_fire_stacks(-20)
 				else if(!HAS_TRAIT(H, TRAIT_NOFIRE))
-					if(environment.gases[/datum/gas/oxygen] && (environment.gases[/datum/gas/oxygen][MOLES]) >= 1) //Same threshhold that extinguishes fire
-						H.adjust_fire_stacks(0.5)
-						if(!H.on_fire && H.fire_stacks > 0)
-							H.visible_message("<span class='danger'>[H]'s body reacts with the atmosphere and bursts into flames!</span>","<span class='userdanger'>Your body reacts with the atmosphere and bursts into flame!</span>")
-						H.IgniteMob()
-						internal_fire = TRUE
+					if(!HAS_TRAIT(H, TRAIT_NO_SELFCOMBUSTION))
```

Comment:
Using a `if(!HAS_TRAIT)` inside another `if(!HAS_TRAIT)` is making me want to punch myself, but the other ways I tried simply failed to work properly so again, feedback would be nice please.

Reply:
Actually, this is a good find. Instead of the new trait, `TRAIT_NO_SELFCOMBUSTION` just give all the mobs `TRAIT_NOFIRE`, and there's already logic for handling this for plasmamen.

Reply:
Should I keep the NO_SELFCOMBUSTION for Highlanders and future use tho? Or just make Highlanders be fireproof too and yeet the NO_SELFCOMBUSTION trait?